gradio/components/dropdown.py

```diff
@@ -41,7 +41,7 @@ class Dropdown(
 
     def __init__(
         self,
-        choices: list[str] | None = None,
+        choices: list[str | tuple[str, str]] | None = None,
```

Comment:
I'm not sure I understand -- what would `choices` being a list of tuple/list/dict look like here?

Reply:
Sorry I mean the second item in the tuple. You could pass around a more complex structure to avoid doing a lookup or something in the predict fn.

Reply:
Ah very interesting point

Reply:
It would require a bit of refactoring to support arbitrary types for the values because we currently pass the values to the frontend and use them in a few different places (e.g. we dispatch the events with the value of the selected choice)

Reply:
For now, let me broaden it up to make sure it works for `string | number` to be consistent with the Radio, Checkboxgroup components. We can revisit arbitrary types if there's demand